invenio/modules/deposit/testsuite/test_deposit_models.py

```diff
@@ -57,9 +63,13 @@ def test_create(self):
         d = Deposition.create(user)
         assert d.type == self.DefaultType
         assert Deposition.get(d.id).type == self.DefaultType
-        d = Deposition.create(user, type=self.AnotherType)
-        assert d.type == self.AnotherType
-        assert Deposition.get(d.id).type == self.AnotherType
+        d2 = Deposition.create(user, type=self.AnotherType)
+        assert d2.type == self.AnotherType
+        assert Deposition.get(d2.id).type == self.AnotherType
+
+        # remove the records
+        Deposition.delete(d)
```

Comment:
Could you please add a test case that checks if both all the underlying BibWorkflowObjects (multiple) and the associated Workflow object has been removed and nothing more than that.